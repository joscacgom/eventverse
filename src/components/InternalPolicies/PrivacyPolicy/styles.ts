import styled from 'styled-components'

export const PrivacyPolicyContainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
    width: 100%;
    height: 100%;
    padding: 0 20px;
    text-align: center;
    font-size: 1.5rem;
    margin-top: 20px;
    & > *:first-child {
        margin: 0 auto;
    }
`
export const PrivacyPolicyTitle = styled.h1`
    width: auto; 
    height: auto; 
    white-space: pre;
    z-index: 1;
    font-weight: 600;
    font-style: normal;
    font-family: "Inter-SemiBold", "Inter", sans-serif;
    color: #000000;
    font-size: 36px;

    @media (max-width: 768px) {
        font-size: 24px;
    }
`

export const PrivacyPolicyText = styled.div`
    margin-bottom: 15px;
    font-size: 1.2rem;
    text-align: justify;

    @media (max-width: 768px) {
        font-size: 0.8rem;
        text-align: justify;
    }
`

export const PrivacyPolicyParagraph = styled.p`
    margin-bottom: 15px;
    font-size: 1.2rem;
`

export const PrivacyPolicyHeader = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
     & > *:first-child {
        margin-bottom: 15px;
    }

`

export const PrivacyPolicyBody = styled.p`
    margin-bottom: 15px;
    font-size: 1.2rem;
`
