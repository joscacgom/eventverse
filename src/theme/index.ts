import { ThemeContext } from 'styled-components'

import { useContext } from 'react'

const nord = {
  black0: '#222222',
  black1: '#434c5e',

  gray0: '#FCFCFC',
  gray1: '#ECEFF4',
  gray2: '#575757',
  gray3: '#B3B3B3',

  red: '#D65151',
  blue: '#0099FF',
  green: '#539362',
  white: '#FFFFFF'
}

const theme = {
  nord
}

export type Theme = typeof theme;

export default theme as Theme

declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}

export const useTheme = () => useContext(ThemeContext)
