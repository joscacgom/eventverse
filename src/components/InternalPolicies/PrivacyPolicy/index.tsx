import React from 'react'
import { PrivacyPolicyContainer, PrivacyPolicyHeader, PrivacyPolicyBody, PrivacyPolicyTitle, PrivacyPolicyText, PrivacyPolicyParagraph } from './styles'
const PrivacyPolicy = () => {
  return (
       <PrivacyPolicyContainer>
            <PrivacyPolicyHeader>
                <PrivacyPolicyTitle>Política de Privacidad</PrivacyPolicyTitle>
            </PrivacyPolicyHeader>
            <PrivacyPolicyBody data-testid="privacy-policy-body">
                <PrivacyPolicyText>
                    <PrivacyPolicyParagraph>
                        El presente Política de Privacidad establece los términos en que EventVerse usa y protege la información que es proporcionada por sus usuarios al momento de utilizar su sitio web. Esta aplicación está comprometida con la seguridad de los datos de sus usuarios. Cuando le pedimos llenar los campos de información personal con la cual usted pueda ser identificado, lo hacemos asegurando que sólo se empleará de acuerdo con los términos de este documento. Sin embargo, esta Política de Privacidad puede cambiar con el tiempo o ser actualizada, por lo que le recomendamos y enfatizamos revisar continuamente esta página para asegurarse de que está de acuerdo con dichos cambios.
                    </PrivacyPolicyParagraph>
                </PrivacyPolicyText>
            </PrivacyPolicyBody>
            <PrivacyPolicyHeader>
                <PrivacyPolicyTitle>Información que se recoge</PrivacyPolicyTitle>
            </PrivacyPolicyHeader>
            <PrivacyPolicyBody>
                <PrivacyPolicyText>
                    <PrivacyPolicyParagraph>
                        Nuestro sitio web puede recopilar información personal, como su nombre, información de contacto (por ejemplo, su dirección de correo electrónico). Ningun otro tipo de dato es recopilado por EventVerse.
                    </PrivacyPolicyParagraph>
                </PrivacyPolicyText>
            </PrivacyPolicyBody>
            <PrivacyPolicyHeader>
                <PrivacyPolicyTitle>Información de terceras partes</PrivacyPolicyTitle>
            </PrivacyPolicyHeader>
            <PrivacyPolicyBody>
                <PrivacyPolicyText>
                    <PrivacyPolicyParagraph>
                        Nuestro sitio web le permite crear una cuenta e iniciar sesión para usar el Servicio a través de los Servicios de redes sociales de terceros soportados en la aplicación.
                        Si decide registrarse mediante dichos servicios, podemos recopilar datos personales que ya están asociados con la cuenta de Su Servicio de redes sociales de terceros, como su nombre, su dirección de correo electrónico y su foto de perfil.
                        Ningun dato extra de su cuenta de redes sociales será compartido con EventVerse.
                    </PrivacyPolicyParagraph>
                </PrivacyPolicyText>
            </PrivacyPolicyBody>
            <PrivacyPolicyHeader>
                <PrivacyPolicyTitle>Uso de la información recopilada</PrivacyPolicyTitle>
            </PrivacyPolicyHeader>
            <PrivacyPolicyBody>
                <PrivacyPolicyText>
                    <PrivacyPolicyParagraph>
                        Nuestro sitio web utiliza la información recopilada para proporcionar el mejor servicio posible,
                        especialmente para mantener un registro de usuarios, y para mejorar nuestros productos y
                        servicios. Es posible que se envíen correos electrónicos por cada compra realizada en la aplicación,
                        nuevos eventos y otra información publicitaria que consideremos relevante para usted o que pueda brindarle algún beneficio.
                        Estos correos electrónicos se enviarán a la dirección que usted proporcione y podrán ser cancelados en cualquier momento.
                        EventVerse está altamente comprometido para cumplir con el compromiso de mantener su información segura.
                        Usamos los sistemas más avanzados y los actualizamos constantemente para asegurarnos que no exista ningún acceso no autorizado.
                    </PrivacyPolicyParagraph>
                </PrivacyPolicyText>
            </PrivacyPolicyBody>
             <PrivacyPolicyHeader>
                <PrivacyPolicyTitle>Cookies</PrivacyPolicyTitle>
            </PrivacyPolicyHeader>
            <PrivacyPolicyBody>
                <PrivacyPolicyText>
                    <PrivacyPolicyParagraph>
                       Una cookie se refiere a un fichero que es enviado con la finalidad de solicitar permiso para almacenarse en su ordenador, al aceptar dicho fichero se crea y la cookie sirve entonces para tener información respecto al tráfico web, y también facilita las futuras visitas a una web recurrente. Otra función que tienen las cookies es que con ellas las web pueden reconocerte individualmente y por tanto brindarte el mejor servicio personalizado de su web.
                       Nuestro sitio web emplea las cookies para poder guardar la sesión del usuario que se registra e inicia sesión en nuestra plataforma. Esta información es empleada únicamente para ofrecer la mejor experiencia de usuario posible. Ningún otro dato es recopilado por EventVerse.
                    </PrivacyPolicyParagraph>
                </PrivacyPolicyText>
            </PrivacyPolicyBody>

            <PrivacyPolicyHeader>
                <PrivacyPolicyTitle>Enlaces a Terceros</PrivacyPolicyTitle>
            </PrivacyPolicyHeader>
            <PrivacyPolicyBody>
                <PrivacyPolicyText>
                    <PrivacyPolicyParagraph>
                      Este sitio web pudiera contener enlaces a otros sitios que pudieran ser de su interés. Una vez que usted de clic en estos enlaces y abandone nuestra página, ya no tenemos control sobre al sitio al que es redirigido y por lo tanto no somos responsables de los términos o privacidad ni de la protección de sus datos en esos otros sitios terceros. Dichos sitios están sujetos a sus propias políticas de privacidad por lo cual es recomendable que los consulte para confirmar que usted está de acuerdo con estas.
                    </PrivacyPolicyParagraph>
                </PrivacyPolicyText>
            </PrivacyPolicyBody>

            <PrivacyPolicyHeader>
                <PrivacyPolicyTitle>Control de su información personal</PrivacyPolicyTitle>
            </PrivacyPolicyHeader>
            <PrivacyPolicyBody>
                <PrivacyPolicyText>
                    <PrivacyPolicyParagraph>
                      Esta compañía no venderá, cederá ni distribuirá la información personal que es recopilada sin su consentimiento, salvo que sea requerido por un juez con un orden judicial.
                      EventVerse Se reserva el derecho de cambiar los términos de la presente Política de Privacidad en cualquier momento.
                    </PrivacyPolicyParagraph>
                </PrivacyPolicyText>
            </PrivacyPolicyBody>
        </PrivacyPolicyContainer>
  )
}

export default PrivacyPolicy
