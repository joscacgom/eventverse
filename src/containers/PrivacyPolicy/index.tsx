import React from 'react'
import { PrivacyPolicy as PrivacyPolicyComponent } from '@/components/InternalPolicies'
import { MainContainer } from './styles'
const PrivacyPolicy = () => {
  return (
    <MainContainer>
     <PrivacyPolicyComponent />
    </MainContainer>
  )
}

export default PrivacyPolicy
