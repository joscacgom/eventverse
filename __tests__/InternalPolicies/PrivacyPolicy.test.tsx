import React from 'react'
/* eslint-disable no-undef */
import 'jest-styled-components'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import theme from '@/theme'
import { ThemeProvider } from 'styled-components'
import PrivacyPolicy from '@/components/InternalPolicies/PrivacyPolicy'

describe('PrivacyPolicy Component', () => {
  it('renders without error', () => {
    render(
        <ThemeProvider theme={theme}>
            <PrivacyPolicy />
        </ThemeProvider>)
  }
  )

  it('displays the privacy policy title', () => {
    render(
        <ThemeProvider theme={theme}>
            <PrivacyPolicy />
        </ThemeProvider>)
    const privacyPolicyTitle = screen.getByText('Política de Privacidad')
    expect(privacyPolicyTitle).toBeInTheDocument()
  })

  it('renders the privacy policy text with the correct content', () => {
    render(<ThemeProvider theme={theme}>
            <PrivacyPolicy />
        </ThemeProvider>)
    const privacyPolicyText = screen.getByText(
      'El presente Política de Privacidad establece los términos en que EventVerse usa y protege la información que es proporcionada por sus usuarios al momento de utilizar su sitio web. Esta aplicación está comprometida con la seguridad de los datos de sus usuarios. Cuando le pedimos llenar los campos de información personal con la cual usted pueda ser identificado, lo hacemos asegurando que sólo se empleará de acuerdo con los términos de este documento. Sin embargo, esta Política de Privacidad puede cambiar con el tiempo o ser actualizada, por lo que le recomendamos y enfatizamos revisar continuamente esta página para asegurarse de que está de acuerdo con dichos cambios.'
    )
    expect(privacyPolicyText).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { container } = render(<ThemeProvider theme={theme}>
            <PrivacyPolicy />
        </ThemeProvider>)
    expect(container).toMatchSnapshot()
  })

  it('renders the privacy policy body', () => {
    render(<ThemeProvider theme={theme}>
            <PrivacyPolicy />
        </ThemeProvider>)
    const privacyPolicyBody = screen.getByTestId('privacy-policy-body')
    expect(privacyPolicyBody).toBeInTheDocument()
  })

  it('displays the correct privacy policy title', () => {
    render(<ThemeProvider theme={theme}>
            <PrivacyPolicy />
        </ThemeProvider>)
    const privacyPolicyTitle = screen.getByText('Política de Privacidad')
    expect(privacyPolicyTitle).toBeInTheDocument()
  })

  it('displays the cookie policy section', () => {
    render(<ThemeProvider theme={theme}>
            <PrivacyPolicy />
        </ThemeProvider>)
    const cookiePolicySection = screen.getByText('Cookies')
    expect(cookiePolicySection).toBeInTheDocument()
  })

  it('displays the third-party information section', () => {
    render(<ThemeProvider theme={theme}>
            <PrivacyPolicy />
        </ThemeProvider>)
    const thirdPartySection = screen.getByText('Información de terceras partes')
    expect(thirdPartySection).toBeInTheDocument()
  })

  it('displays the control of personal information section', () => {
    render(<ThemeProvider theme={theme}>
            <PrivacyPolicy />
        </ThemeProvider>)
    const controlSection = screen.getByText('Control de su información personal')
    expect(controlSection).toBeInTheDocument()
  })
})
