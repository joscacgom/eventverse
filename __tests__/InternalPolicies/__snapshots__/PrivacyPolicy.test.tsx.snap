// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyPolicy Component matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  padding: 0 20px;
  text-align: center;
  font-size: 1.5rem;
  margin-top: 20px;
}

.c0 > *:first-child {
  margin: 0 auto;
}

.c2 {
  width: auto;
  height: auto;
  white-space: pre;
  z-index: 1;
  font-weight: 600;
  font-style: normal;
  font-family: "Inter-SemiBold","Inter",sans-serif;
  color: #000000;
  font-size: 36px;
}

.c4 {
  margin-bottom: 15px;
  font-size: 1.2rem;
  text-align: justify;
}

.c5 {
  margin-bottom: 15px;
  font-size: 1.2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 > *:first-child {
  margin-bottom: 15px;
}

.c3 {
  margin-bottom: 15px;
  font-size: 1.2rem;
}

@media (max-width:768px) {
  .c2 {
    font-size: 24px;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 0.8rem;
    text-align: justify;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Política de Privacidad
      </h1>
    </div>
    <div
      class="c3"
      data-testid="privacy-policy-body"
    >
      <div
        class="c4"
      >
        <p
          class="c5"
        >
          La presente Política de Privacidad establece los términos en que EventVerse usa y protege la información que es proporcionada por sus usuarios al momento de utilizar su sitio web. Esta aplicación está comprometida con la seguridad de los datos de sus usuarios. Cuando le pedimos llenar los campos de información personal con la cual usted pueda ser identificado, lo hacemos asegurando que sólo se empleará de acuerdo con los términos de este documento. Sin embargo, esta Política de Privacidad puede cambiar con el tiempo o ser actualizada, por lo que le recomendamos y enfatizamos revisar continuamente esta página para asegurarse de que está de acuerdo con dichos cambios.
        </p>
      </div>
    </div>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Información que se recoge
      </h1>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5"
        >
          Nuestro sitio web puede recopilar información personal, como su nombre, información de contacto (por ejemplo, su dirección de correo electrónico). Ningun otro tipo de dato es recopilado por EventVerse.
        </p>
      </div>
    </div>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Información de terceras partes
      </h1>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5"
        >
          Nuestro sitio web le permite crear una cuenta e iniciar sesión para usar el Servicio a través de los Servicios de redes sociales de terceros soportados en la aplicación. Si decide registrarse mediante dichos servicios, podemos recopilar datos personales que ya están asociados con la cuenta de Su Servicio de redes sociales de terceros, como su nombre, su dirección de correo electrónico y su foto de perfil. Ningun dato extra de su cuenta de redes sociales será compartido con EventVerse.
        </p>
      </div>
    </div>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Uso de la información recopilada
      </h1>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5"
        >
          Nuestro sitio web utiliza la información recopilada para proporcionar el mejor servicio posible, especialmente para mantener un registro de usuarios, y para mejorar nuestros productos y servicios. No es posible que se envíen correos electrónicos por cada compra realizada en la aplicación, nuevos eventos u otra información publicitaria. EventVerse está altamente comprometido para cumplir con el compromiso de mantener su información segura.
        </p>
      </div>
    </div>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Cookies
      </h1>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5"
        >
          Una cookie se refiere a un fichero que es enviado con la finalidad de solicitar permiso para almacenarse en su ordenador, al aceptar dicho fichero se crea y la cookie sirve entonces para tener información respecto al tráfico web, y también facilita las futuras visitas a una web recurrente. Otra función que tienen las cookies es que con ellas las web pueden reconocerte individualmente y por tanto brindarte el mejor servicio personalizado de su web. Nuestro sitio web emplea las cookies para poder guardar la sesión del usuario que se registra e inicia sesión en nuestra plataforma. Esta información es empleada únicamente para ofrecer la mejor experiencia de usuario posible. Ningún otro dato es recopilado por EventVerse.
        </p>
      </div>
    </div>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Enlaces a Terceros
      </h1>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5"
        >
          Este sitio web pudiera contener enlaces a otros sitios que pudieran ser de su interés. Una vez que usted de clic en estos enlaces y abandone nuestra página, ya no tenemos control sobre al sitio al que es redirigido y por lo tanto no somos responsables de los términos o privacidad ni de la protección de sus datos en esos otros sitios terceros. Dichos sitios están sujetos a sus propias políticas de privacidad por lo cual es recomendable que los consulte para confirmar que usted está de acuerdo con estas.
        </p>
      </div>
    </div>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Control de su información personal
      </h1>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5"
        >
          Esta compañía no venderá, cederá ni distribuirá la información personal que es recopilada sin su consentimiento, salvo que sea requerido por un juez con un orden judicial. EventVerse Se reserva el derecho de cambiar los términos de la presente Política de Privacidad en cualquier momento.
        </p>
      </div>
    </div>
  </div>
</div>
`;
